kedro/framework/cli/utils.py

```diff
@@ -318,6 +319,27 @@ def _check_module_importable(module_name: str) -> None:
         ) from exc
 
 
+def get_entry_points(name: str) -> importlib_metadata.EntryPoints:
```

Comment:
```suggestion
def _get_entry_points(name: str) -> importlib_metadata.EntryPoints:
```
The fewer functions we advertise as public the better 😀

Reply:
I think the main issue is we can't control what the plugin does, the slowness is mainly due to loading up tons of python libraries when we don't need it. (e.g. `kedro-viz` dependencies are all loaded even though we are just doing the discovery command `kedro`).

This may be one thing we can consider when we are chopping off some kedro CLI and re-organizing some of them into sub-commands, there are rooms to improve the CLI performance.

For the template plugin - I don't think it should carry many libraries, so I am not too concerned this would slow down the CLI too much.